packages/shared/src/aws/tile.metadata.base.ts

```diff
@@ -190,6 +193,55 @@ export class TileMetadataTableBase {
         return DynamoDB.Converter.unmarshall(item.Item) as T;
     }
 
+    /**
+     * Iterate over the entire table yielding each batch of items
+
+     * @param TableName the name of the table to interate over
+     */
+    async *scan(TableName: string): AsyncGenerator<DynamoDB.Types.ItemList, null, void> {
+        let exclusiveStartKey: DynamoDB.Key | undefined = undefined;
+        while (true) {
+            const items: DynamoDB.Types.ScanOutput = await this.dynamo
+                .scan({
+                    TableName,
+                    ExclusiveStartKey: exclusiveStartKey,
+                })
+                .promise();
+
+            if (items.Items == null) break;
+
+            yield items.Items;
+
+            exclusiveStartKey = items.LastEvaluatedKey;
+            if (exclusiveStartKey == null) break;
+        }
+        return null;
+    }
+
+    /**
+     * Write an array of write requests to the table.
+
+     * @param tableName the name of the table to write to
+
+     * @param input the array of write requests
+
+     * @param maxBatchSize the maximum number of write requests to send in one command
+     */
+    async batchWrite(tableName: string, input: DynamoDB.WriteRequests, maxBatchSize = 25): Promise<void> {
```

Comment:
this should use a `T` in the input

Reply:
I don't understand? template input? Only `WriteRequests` are allowed.

Reply:
This is no longer needed